### ISSUE TITLE
cleanup for "create_tee_file" fucntion

### ISSUE THREAD

**Issue by seungeun2** [OP-TEE/optee_os#554]
hi, i think there is a issue with creating persistent object.

if optee_os fails with creating a persistent object, then next time it creates a persistent object, it cannot create a persistent object with the same object id, because tee_file directory is not properly removed.

To be specific,

Within "create_tee_file" function (core/tee/tee_fs_common.c, line#:1150)

	/* create meta file in TEE file directory */
	meta = create_meta_file(file);
	if (!meta)
		EMSG("Failed to create new meta file");

if "create_meta_file" function fails, then shouldn't it also remove the directory(uuid/obj_id) as a cleanup routine?

**Comment by m943040028** [comment]
Hi @seungeun2

Sorry for late reply. I am extremely busy these days.
You're right. if we leave an empty folder, we will fail to create a persistent object with the same id.
However, remove the directory as a cleanup routine might not completely solve this issue. Think if we have a power cut before we remove the folder. What happened if we failed to remove the folder?

I created https://github.com/OP-TEE/optee_os/pull/581 that addresses this issue.
Feel free to leave comments.
